### PR TITLE
Client log view improvements

### PR DESCRIPTION
**Pull request by DevonRD**
Fixes the following issues with client log view:
- Inconsistent and excessive spacing between log entries
- Lack of responsiveness to window size or scaling

Adds the following features to log view:
- Auto scroll if scrolled to bottom
- Ability to clear logs on button click
- Search function to filter logs
- General styling refinements to increase usability

*A couple files have inconsequential changes from linting/formatting

### CODE REVIEW

**Comment by mcm001** [comment]
The GOAT returns

**Comment by mcm001** [comment]
This reminds me - we should open a ticket to grab log messages from before the UI connects. We do this in a hacky way right now, but only for the first client.

**Comment by srimanachanta** [comment]
This needs to be done in the composable vue API for the transition to Vue3, I already have Spacing, coloring, reactivity, and styling on the vue3 branch. Will PR soon, we should merge this either directly in that PR or after we merge that one.

File: photon-client/src/components/app/photon-log-view.vue [hunk 1]
```diff
@@ -70,58 +86,112 @@ document.addEventListener("keydown", (e) => {
...
+          </v-col>
+          <v-col cols="12" md="7">
+            <v-row class="no-gutters">
+              <v-col v-for="level in [0, 1, 2, 3]" :key="level">
+                <v-row dense align="center">
+                  <v-col cols="6" md="8" style="text-align: right">
+                    {{ getLogLevelFromIndex(level) }}
+                  </v-col>
+                  <v-col cols="6" md="4">
+                    <v-switch v-model="selectedLogLevels[level]" dark color="#ffd843" />
```
Comment (srimanachanta) [inline photon-client/src/components/app/photon-log-view.vue:125]: Whats the reason to replace the button toggle with switches?

**Comment by srimanachanta** [comment]
closes #1385 1384

**Review by mcm001 — APPROVED** [review]
Does it work? I have no idea! Sure Lgtm tho

**Comment by srimanachanta** [comment]
These changes will likely have to be looked at again during the Vue3 changes

**Pull request merged**